test/coercive.rb

```diff
@@ -100,6 +100,33 @@ def assert_coercion_error(errors)
     end
   end
 
+  describe "integer" do
+    before do
+      @coercion = Module.new do
+        extend Coercive
+
+        attribute :foo, integer, required
+        attribute :bar, integer, optional
+      end
+    end
+
+    it "coerces the input value to an integer" do
+      attributes = { "foo" => "100", "bar" => 100 }
+
+      expected = { "foo" => 100, "bar" => 100 }
+
+      assert_equal expected, @coercion.call(attributes)
+    end
```

Comment:
What happens when it gets a value like `100.5` or `"100.5"`? Would be good to have a test for that.

Reply:
Good call, thank you! `Integer("100.5")` fails as expected, but `Integer(100.5)` returns `100`. I'm just going to throw an error if the input is `Float`

Reply:
I just added a check and some tests in a99d5ac714559cfe3187cfbecf2103dadda2c4fa